src/SW_netCDF_Input.c

```diff
@@ -0,0 +1,6230 @@
+}
+
+/**
+@brief Read/store the spatial coordinates that the program contains
+
+@param[out] SW_netCDFIn Constant netCDF input file information
+@param[in] domCoordVarNames A list of the two latitude/y and longitude/x
+coordinate variable/dimension names
+@param[in] siteName User-provided site dimension/variable "site" name
+@param[in] domFileID Domain file identifier
+@param[in] domType Type of domain - "s" or "xy"
+@param[in] primCRSIsGeo Specifies if the current CRS type is geographic
+@param[out] LogInfo Holds information dealing with logfile output
+*/
+static void read_domain_coordinates(
+    SW_NETCDF_IN *SW_netCDFIn,
+    char *domCoordVarNames[],
+    char *siteName,
+    int domFileID,
+    char *domType,
+    Bool primCRSIsGeo,
+    LOG_INFO *LogInfo
+) {
+    int index;
+    int varID;
+    size_t val;
+    const int numReadInDims = (primCRSIsGeo) ? 2 : 4;
+    const int numDims = 2;
+    Bool validY;
+    Bool validX;
+
+    char *domCoordNames[2]; /* Set later */
+
+    size_t *domCoordSizes[] = {
+        &SW_netCDFIn->domYCoordGeoSize,
+        &SW_netCDFIn->domXCoordGeoSize,
+        &SW_netCDFIn->domYCoordProjSize,
+        &SW_netCDFIn->domXCoordProjSize
+    };
+    double **domCoordArrs[] = {
+        &SW_netCDFIn->domYCoordsGeo,
+        &SW_netCDFIn->domXCoordsGeo,
+        &SW_netCDFIn->domYCoordsProj,
+        &SW_netCDFIn->domXCoordsProj
+    };
+
+    /* Determine the dimension lengths from the currect dimension
+       names matching site/xy and geographical/projected */
+    if (strcmp("s", domType) == 0) {
+        domCoordNames[0] = siteName;
+        domCoordNames[1] = siteName;
+    } else {
+        if (primCRSIsGeo) {
+            domCoordNames[0] = domCoordVarNames[0];
+            domCoordNames[1] = domCoordVarNames[1];
+        } else {
+            domCoordNames[0] = domCoordVarNames[2];
+            domCoordNames[1] = domCoordVarNames[3];
+        }
+    }
+
+    for (index = 0; index < numDims; index++) {
+        SW_NC_get_dimlen_from_dimname(
+            domFileID, domCoordNames[index], domCoordSizes[index], LogInfo
+        );
+        if (LogInfo->stopRun) {
+            return; /* Exit function prematurely due to error */
+        }
+
+        /* Match the same for projected sizes (may not be used) */
+        *(domCoordSizes[index + 2]) = *(domCoordSizes[index]);
+    }
+
+    alloc_dom_coord_info(domCoordArrs, domCoordSizes, numReadInDims, LogInfo);
+    if (LogInfo->stopRun) {
+        return; /* Exit function prematurely due to error */
+    }
+
+    for (index = 0; index < numReadInDims; index++) {
+        varID = -1;
+
+        SW_NC_get_vals(
+            domFileID,
+            &varID,
+            domCoordVarNames[index],
+            *domCoordArrs[index],
+            LogInfo
+        );
+        if (LogInfo->stopRun) {
+            return; /* Exit prematurely due to error */
+        }
+
+        for (val = 0; val < *domCoordSizes[index]; val++) {
+            /* Check latitude */
```

Comment:
read_domain_coordinates() [as commit d09004e0d1e818b19eb5cdd96fb942a90c0a96ae] loops through all coordinate variables and checks variables associated with the Y-axis dimension to be within [-90, 90] and variables associated with the X-axis dimension to be within [-180, 180].

There are two problems with the current version
* The limits [-90,90] and [-180,180] are only valid for geographic but not projected coordinate variables --> skip these checks for projected coordinate variables.
* There are two conventions for longitude [-180,180] and [0,360], and more generally, longitude can "wrap" around (i.e., `(180 + lon % 360) - 180`) --> don't check longitude values.

Thus, the only remaining check would be for the geographic Y-axis variable latitude. Not sure the for-loop is warranted. 

Thanks!